addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/AjaxSpiderTarget.java

```diff
@@ -275,7 +314,24 @@ public AjaxSpiderTarget build() {
                 throw new IllegalStateException("The starting URI is not in scope.");
             }
 
-            return new AjaxSpiderTarget(startUri, inScopeOnly, context, user, options, subtreeOnly);
+            List<ExcludedElement> effectiveExcludedElements;
+            if (excludedElements != null) {
+                effectiveExcludedElements = excludedElements;
+            } else if (context != null && extension != null) {
+                effectiveExcludedElements =
+                        extension.getContextDataManager().getExcludedElements(context);
+            } else {
+                effectiveExcludedElements = List.of();
```

Comment:
Does this work out the same to the places we used to use `Collections.emptyList()`?

Reply:
Yes.